### DISCUSSION TITLE
[Support] 加了中间件之后 异常无法捕获，阻断运行

### DISCUSSION THREAD

**Discussion by hjane** [midwayjs/midway#3684] [Q&A]
### Describe the problem（描述问题）

配置了  koa-sse-stream 如下图
<img width="842" alt="image" src="https://github.com/midwayjs/midway/assets/2201027/3b3e9720-736d-49c2-b48c-8168644a5157">

如果流报错，会出现错误，阻断运行
<img width="853" alt="image" src="https://github.com/midwayjs/midway/assets/2201027/541b0e74-946b-4653-8cf9-3c85561f0053">

请问如何处理才能不阻断运行？

<img width="748" alt="image" src="https://github.com/midwayjs/midway/assets/2201027/2dc24ec1-dc46-48d4-a1d4-0592ccdd38de">

### Midway Versions（Midway 版本）

 "@midwayjs/bootstrap": "^3.12.0",
    "@midwayjs/cache-manager": "^3.14.7",
    "@midwayjs/core": "^3.12.0",
    "@midwayjs/decorator": "^3.12.0",
    "@midwayjs/http-proxy": "^3.14.9",
    "@midwayjs/info": "^3.12.0",
    "@midwayjs/koa": "^3.12.0",
    "dotenv": "^16.4.1",
    "googleapis": "^133.0.0",
    "install": "^0.13.0",
    "koa-sse-stream": "^0.2.0",
    "openai": "^4.28.0",
    "short-uuid": "^4.2.2",
    "stream": "^0.0.2",

**Comment by czy88840616** [comment]
流不会执行中间件，无法通过中间件处理，需要自行保证正确性。

**Reply by hjane** [reply]
非常感谢您的解答！我还想问一下，我通过 stream.on("error", err => {.  }) 监听到了错误，如果我不想出现阻断的错误，或者在程序报错后能自动恢复，有什么办法吗？或者给我一个方向，我可以自己去找找资料。

**Comment by czy88840616** [comment]
流错误需要在流内部完全处理掉，这就不会阻断影响程序了。
